File: 09_methods/main.go
package main

import "fmt"

// Fruit is a sample struct.
type Fruit struct {
	name  string
	count int
}

func (f Fruit) print() {
	fmt.Println("name =", f.name, " count =", f.count)
}

func main() {
	a := Fruit{name: "Apple", count: 12}
	a.print()
}

File: 09_methods/main.js
class Fruit {
  constructor(name, count) {
    this.name = name;
    this.count = count;
  }

  print() {
    console.log({
      name: this.name,
      count: this.count
    });
  }
}

let a = new Fruit("Apple", 12);
a.print();
